File: Codewars/5-solution.js
// Codewars 5
// Title : Reversed Strings
// Instructions : Complete the solution so that it reverses the string passed into it.
// Challange Url : https://www.codewars.com/kata/5168bb5dfe9a00b126000018/
// Level : 8kyu

// Method : 1
// function solution(str) {
//   const result = [];
//   const arrStr = str.split("");
//   for (let i = arrStr.length; i >= 0; i--) {
//     result.push(arrStr[i]);
//   }
//   return result.join("");
// }

// Method : 2
const solution = (str) => str.split("").reverse().join("");

// Other Users Method

// Result ot debug
console.log(solution("world"));
console.log(solution("hello"));
console.log(solution(""));
console.log(solution("h"));

File: Codewars/python/8-kyu/5-solution.py
# Codewars 5
# Title : Reversed string
# Question : Complete the solution so that it reverses the string passed into it. 
# Challange Url : https://www.codewars.com/kata/5168bb5dfe9a00b126000018
# Level : 8kyu

# Method : 1
def solution(string):
    if(string == ''): return ''
    reversed_string = []
    array_string = list(string)
    for i,c in enumerate(array_string):
        reversed_string.append(array_string[len(string) - i - 1])  
    return ''.join(reversed_string)

# Method : 2
def solution(s): return ''.join(reversed(s))

# Other Users Method
def solution(str):
  return str[::-1]

# Log or debug
print(solution('world'))
print(solution('hello'))
print(solution(''))
print(solution('h'))

